logging/MonologFactory.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+namespace SilverStripe\Framework\Logging;
+
+/**
+ * Adaptor connecting Monolog to SilverStripe for error handling
+ */
+class MonologFactory extends \InjectionCreator
+{
+	public function create($class, array $params = array()) {
+		// Register the default error handler
+		$logger = parent::create($class, $params);
+
+		// Load the error handlers into the monolog loader
+		$injector = \Injector::inst();
+
+		$handlerID = null;
+		$handlers = \Config::inst()->get('Logger', 'handlers');
+		foreach($handlers as $handler) {
+			// A hacky mechanism for disabling handlers in yaml config
+			if(in_array(strtolower($handler['class']), array('none', 'disable'))) {
+				continue;
+			}
+
+			// Special treatment for Formatter
+			if(isset($handler['properties']['Formatter']['class'])) {
```

Comment:
Why not just make the handlers a dependency of MonologFactory itself, like how filters is a dependency of RequestProcessor?

```

/** DI this */
protected $handlers = array();

public function create($class, array $params = array()) {
		// Register the default error handler
		$logger = parent::create($class, $params);
		foreach($this->getHandlers() as $handler) {
			$logger->pushHandler($handler);
		}
		return $logger;
	}
```